src/main/java/seedu/address/storage/JsonAddressBookStorage.java

```diff
@@ -4,77 +4,214 @@
         requireNonNull(addressBook);
         requireNonNull(filePath);
 
+        // Creates the file if it does not exist.
         FileUtil.createIfMissing(filePath);
+        // Serializes the address book and saves it to the file.
         JsonUtil.saveJsonFile(new JsonSerializableAddressBook(addressBook), filePath);
     }
 
+    /**
+     * Saves the provided list of PropertyForRent to the file path.
+     *
+     * @param properties the list of PropertyForRent to save.
+     * @throws IOException if there is an error saving the properties.
+     */
+    @Override
+    public void savePropertiesForRent(List<PropertyForRent> properties) throws IOException {
+        requireNonNull(properties);
+        // Converts the list of PropertyForRent to a list of JsonAdaptedPropertyForRent.
+        List<JsonAdaptedPropertyForRent> jsonProperties = properties.stream()
+                .map(JsonAdaptedPropertyForRent::new)
+                .collect(Collectors.toList());
+        JsonSerializableAddressBook jsonAddressBook = new JsonSerializableAddressBook();
+        // Sets the properties for rent in the JsonSerializableAddressBook.
+        jsonAddressBook.setPropertiesForRent(jsonProperties);
+        // Saves the JsonSerializableAddressBook to the file.
+        JsonUtil.saveJsonFile(jsonAddressBook, filePath);
+    }
+
+    /**
+     * Saves the provided list of PropertyForSale to the file path.
+     *
+     * @param properties the list of PropertyForSale to save.
+     * @throws IOException if there is an error saving the properties.
+     */
+    @Override
+    public void savePropertiesForSale(List<PropertyForSale> properties) throws IOException {
+        requireNonNull(properties);
+        // Converts the list of PropertyForSale to a list of JsonAdaptedPropertyForSale.
+        List<JsonAdaptedPropertyForSale> jsonProperties = properties.stream()
+                .map(JsonAdaptedPropertyForSale::new)
+                .collect(Collectors.toList());
+        JsonSerializableAddressBook jsonAddressBook = new JsonSerializableAddressBook();
+        // Sets the properties for sale in the JsonSerializableAddressBook.
+        jsonAddressBook.setPropertiesForSale(jsonProperties);
+        // Saves the JsonSerializableAddressBook to the file.
+        JsonUtil.saveJsonFile(jsonAddressBook, filePath);
+    }
+
+    /**
+     * Reads properties for rent from the JSON file and returns a list of PropertyForRent.
+     *
+     * @return a list of PropertyForRent read from the file.
+     * @throws DataLoadingException if there is an error loading properties for rent.
+     */
+    public List<PropertyForRent> readPropertiesForRent() throws DataLoadingException {
+        JsonSerializableAddressBook jsonAddressBook = JsonUtil.readJsonFile(filePath, JsonSerializableAddressBook.class)
+                .orElseThrow(() -> new RuntimeException("Failed to load properties for rent"));
```

Comment:
Same as previous, turn strings into constants for this and the 2 other strings in the method